ArcGISRuntimeSDKQt_CppSamples/Scenes/OpenMobileScenePackage/OpenMobileScenePackage.cpp

```diff
@@ -0,0 +1,153 @@
+// [WriteFile Name=OpenMobileScenePackage, Category=Scenes]
+// [Legal]
+// Copyright 2018 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "OpenMobileScenePackage.h"
+
+#include "MobileScenePackage.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+
+#include <QDir>
+#include <QTemporaryDir>
+#include <QtCore/qglobal.h>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif // Q_OS_IOS
+
+using namespace Esri::ArcGISRuntime;
+
+OpenMobileScenePackage::OpenMobileScenePackage(QObject* parent /* = nullptr */):
+  QObject(parent)
+{  
+}
+
+OpenMobileScenePackage::~OpenMobileScenePackage() = default;
+
+// helper method to get cross platform data path
+namespace {
+  QString defaultDataPath()
+  {
+    QString dataPath;
+
+  #ifdef Q_OS_ANDROID
+    dataPath = "/sdcard";
+  #elif defined Q_OS_IOS
+    dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+  #else
+    dataPath = QDir::homePath();
+  #endif
+
+    return dataPath;
+  }
+}
+
+void OpenMobileScenePackage::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<OpenMobileScenePackage>("Esri.Samples", 1, 0, "OpenMobileScenePackageSample");
+}
+
+SceneQuickView* OpenMobileScenePackage::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void OpenMobileScenePackage::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+  {
+    return;
+  }
+
+  m_sceneView = sceneView;
+
+  openPackage();
+
+  emit sceneViewChanged();
+}
+
+// Open a mobile scene package
+void OpenMobileScenePackage::openPackage()
+{
+  // create the MSPK data path
+  const QString dataPath = defaultDataPath() + "/ArcGIS/Runtime/Data/mspk/philadelphia.mspk";
```

Comment:
I think in samples like this one where the data must be downloaded, it would be nice to add a comment to that effect for users who are running these stand-alone